File: LMProject/src/main/java/fpt/aptech/LMProject/controller/BookController.java
package fpt.aptech.LMProject.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

import fpt.aptech.LMProject.services.BookService;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;

import fpt.aptech.LMProject.entities.Book;

@RestController
@Slf4j
@AllArgsConstructor
public class BookController {

    private BookService bookService;

    @GetMapping("/api/book/{id}")
    public Book getBook(@PathVariable(name = "id") int bookId) {
        log.info("Receiving request get all Book .....!!!!!");
        return bookService.getBook(bookId);
    }


}

File: backend/src/main/kotlin/com/example/kotlindemo/controller/BookController.kt
package com.example.kotlindemo.controller

import com.example.kotlindemo.model.Book
import com.example.kotlindemo.repository.BookRepository
import org.springframework.web.bind.annotation.*
import javax.validation.Valid


@RestController
@RequestMapping("/api/books")
class BookController(private val bookRepository: BookRepository) {
  @GetMapping("/all")
  fun getAllBooks(): List<Book> {
    return bookRepository.findAll()
  }

  @PostMapping("/add")
  fun createNewBook(@Valid @RequestBody book: Book): Book {
    return bookRepository.save(book)
  }

  @GetMapping("/detail/{id}")
  fun getBookById(@PathVariable id: Long): Book {
    return bookRepository.findById(id).get()
  }

  @PutMapping("/update/{id}")
  fun updateBookById(@PathVariable id: Long, @RequestBody book: Book): Book {
    val bookToUpdate = bookRepository.findById(id).get()
    book.bookid = bookToUpdate.bookid

    return bookRepository.save(book)
  }

  @DeleteMapping("/delete/{id}")
  fun deleteBookById(@PathVariable id: Long): String {
    bookRepository.deleteById(id)
    return "Book deleted successfully"
  }
}
